hazelcast-spring/src/main/java/com/hazelcast/spring/HazelcastConfigBeanDefinitionParser.java

```diff
@@ -335,6 +339,18 @@ private AbstractBeanDefinition handleService(Node node) {
                     }
                 }
             }
+
+            final NamedNodeMap attrs = node.getAttributes();
+            Node classNameNode = attrs.getNamedItem("class-name");
+            if (classNameNode != null) { 
+            	serviceConfigBuilder.addPropertyValue("className", getTextContent(classNameNode));
+            }
+            Node implNode = attrs.getNamedItem("implementation");
+            if (implNode != null) { 
+                serviceConfigBuilder.addPropertyReference("implementation", getTextContent(implNode));
+            }
+            Assert.isTrue(classNameSet || classNameNode != null || implNode != null, "One of 'class-name' or 'implementation' "
```

Comment:
Ah, I see now. I left the `class-name` element in schema for backward compatibility. It would be nice to specify it as deprecated somewhere in docs, and, probably, in the XSD itself. So if the old `class-name` element is set, the configuration is still valid. as well as `class-name` or `implementation` attribute. The proper way would be to remove the `class-name` element at all, but this will break backward compatibility. But this is a common case for major releases (3.6 -> 3.7), actually.